### ISSUE TITLE
Filter-object of b-table not reactive anymore

### ISSUE THREAD

**Issue by FabianAlbrecht** [bootstrap-vue/bootstrap-vue#4065]
### Describe the bug

When you pass a filter-object to b-table the filter-function will not be triggered when the properties of the object change. (No reactivity)
This happens since version 2.0.0.

### Steps to reproduce the bug

1. Open Demo-Link https://codesandbox.io/s/gallant-curie-dkrsc
2. Change Text in the Filter -> Nothing happens
3. Change the version of bootstrap-vue to 2.0.0-rc.28
4. Change Text in the Filter -> Table is correctly filtered

### Expected behavior

The Table is filtered according to the entered value

### Versions

**Libraries:**

- BootstrapVue: 2.0.0
- Bootstrap: 4.3.1
- Vue: 2.6.10

**Environment:**

- Device: Mac
- OS: macOS Mojave
- Browser: Chrome
- Version: 76

### Demo link

https://codesandbox.io/s/gallant-curie-dkrsc

**Comment by caiquecastro** [comment]
Added a test on #4067 to confirm this regression.

**Comment by tmorehouse** [comment]
Props are typically "watched" by Vue at the reference layer.  We would need to implement a deep watcher to look for property changes.

When generating your filter object, use a computed property that returns a new object when one of the children properties changes:

```js
computed: {
  filterCriteria() {
    return {
      a: this.a,
      b: this.b
    }
  }
}
```

**Comment by tmorehouse** [comment]
@caiquecastro it is possible that providers may be called twice (initially) if you don't have a field definition provided (so b-table can lookup the field names)

**Comment by tmorehouse** [comment]
@FabianAlbrecht there is a mistake in your template.

You are passing a string to the `fields` prop:

````html
<template>
  <div>
    <b-form-input v-model="filterObject.filterText" />
    <b-table :items="items" fields='name' :filter="filterObject" :filter-function="filterFunction" />
  </div>
</template>
````

You need to pass an array of field names:

```html
<template>
  <div>
    <b-form-input v-model="filterObject.filterText" />
    <b-table :items="items" :fields="['name']" :filter="filterObject" :filter-function="filterFunction" />
  </div>
</template>
```

**Comment by FabianAlbrecht** [comment]
@tmorehouse You are right, that i made a mistake here.
I have fixed this in my Codesandbox (https://codesandbox.io/s/gallant-curie-dkrsc). But this does not fix the problem for me.
Even in the Codesandbox you posted the filtering does not work for me.

**Comment by tmorehouse** [comment]
If I type in `1` into the filter, it shows only `Item 1`

**Comment by FabianAlbrecht** [comment]
Unfortunately not for me. Maybe there is a problem with the Sandbox?

When I open your sandbox (https://codesandbox.io/s/quizzical-haslett-9jfr7) it shows the following error in the console, because there is a colon missing in front of the 'fields' prop:
`[Vue warn]: Invalid prop: type check failed for prop "fields". Expected Array, got String with value "['name']".`

Is it possible that it is not the current version of your code?

**Comment by tmorehouse** [comment]
I might not have saved it when forking....

**Comment by tmorehouse** [comment]
When using a computed prop to return the filter object (which generates a new filter object whenever the criteria changes) it works...

But when passing the same object reference of which properties inside change, it doesn't appear to be detecting the change. Will may need to add in a deep watcher on the `filter` prop

**Comment by tmorehouse** [comment]
PR #4068 fixes the issue...  and will be available in v2.0.1 later this week.

As a workaround, you can use a computed prop to generate the filter object (which creates a new object when the criteria changes:

```html
<template>
  <div>
    <b-form-input v-model="filterObject.filterText" />
    <b-table
       :items="items"
       :fields="['name']"
       :filter="computedFilterObject"
       :filter-function="filterFunction"
     />
  </div>
</template>

<script>
export default {
  data() {
    return {
      items: [{ name: "Item 1" }, { name: "Item 2" }],
      filterObject: { filterText: '' }
    }
  },
  computed: {
    computedFilterObject() {
      return {
        filterText: this.filterObject.filterText
      }
    }
  },
  methods: {
    filterFunction(item, filterObject) {
      console.log(new Date(), item.name, filterObject.filterText);
      return item.name.toLowerCase().includes(filterObject.filterText);
    }
  }
}
</script>
```

**Comment by FabianAlbrecht** [comment]
Thanks for fixing the issue. Your workaround works perfectly.

**Comment by tmorehouse** [comment]
@FabianAlbrecht the fix will be in v2.0.1, which would hopefully be released later this week.
